lib/src/any_map.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2024 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(missing_docs)]
+
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
+
+/// Type-safe, thread-safe map that stores objects of arbitrary types.
```

Comment:
nit: "thread-safe" is no longer relevant here.

It might be also good to inline it, but I don't have strong opinion about that.

Reply:
I think this may see some reuse when I redo https://github.com/martinvonz/jj/pull/3194 so I'd rather leave it as a lib.